src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -126,6 +130,11 @@ namespace ts.refactor.convertParamsToDestructuredObject {
                 }
                 // if the refactored function is a constructor, we must also check if the references to its class are valid
                 if (isConstructor && contains(classSymbols, checker.getSymbolAtLocation(entry.node), symbolComparer)) {

```

Comment:
>checker.getSymbolAtLocation(entry.node) [](start = 60, length = 39)

It looks like we're going to dig through aliases a linear number of times, which isn't ideal.  Another way to structure this would be to dig through the target symbol once, up front, and then search for the actual target symbol.  Of course, it looks like "linear" means "1 or 2", so the extra complexity may not be justified in this case.